src/coreclr/src/jit/optcse.cpp

```diff
@@ -2703,28 +2818,165 @@ class CSE_Heuristic
                 else if (currVN != firstVN)
                 {
                     allSame = false;
-                    break;
+                }
+
+                BasicBlock*                blk       = lst->tslBlock;
+                const BasicBlock::weight_t curWeight = blk->getBBWeight(m_pCompiler);
+                if (isConstCSE)
+                {
+                    // See if we already have this const in our item[] table
+                    bool match = false;
+                    for (i = 0; i < maxItem; i++)
+                    {
+                        if (item[i].constVN == currVN)
+                        {
+                            item[i].totalWeight += curWeight;
+                            match = true;
+                        }
+                    }
+
+                    bool sortAndAdd = false;
+                    if (match == false)
+                    {
+                        if (maxItem < 8)
+                        {
+                            j = maxItem;
+                            // Add the new value as item[j]
+                            //
+                            item[j].constVN     = currVN;
+                            item[j].totalWeight = curWeight;
+                            item[j].constValue  = m_pCompiler->vnStore->CoercedConstantValue<ssize_t>(currVN);
+
+                            if (j == 0)
+                            {
+                                // Prefer to sort the very first entry as highest, so bump up its weight by 1
+                                item[j].totalWeight += 1;
+                            }
+                            maxItem++;
+                        }
+                        else
+                        {
+                            // We will sort item[*] and add this value, replacing the lowest value in item[7]
+                            sortAndAdd = true;
+                        }
+                    }
+
+                    // We always need to perform one sort before finishing
+                    // so we will sort if islastNode is true
+                    //
+                    if (isLastNode || sortAndAdd)
+                    {
+                        // sort the item[8] by totalWeight
```

Comment:
Could you please look at `qsort` usage in jit sources and see if you can use it instead of a new N^2 sort. I do not see any errors in this bubble sort implementation, but I don't see the motivation here.

Reply:
Typically we have very few different candidates that are sharing the same base constant. 
I can change this into a priority heap, which would optimize the number of compares.
qsort isn't going to be faster when we have only 7 elements and they are already mostly sorted.

Reply:
Seems reasonable, though this could at least be factored out.

Reply:
After more investigation, I turns out that there isn't any real benefit by making the the most common case the base (zero-offset)  Instead it is better to chose the lowest value so that the offset are positive.  Larger negative values require a an extra subtract instruction.  So I am able to delete all of the sorting logic.